src/dft/descriptor.cpp

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#include "oneapi/mkl/dft/descriptor.hpp"
+
+#ifdef ENABLE_MKLCPU_BACKEND
+#include "oneapi/mkl/dft/detail/mklcpu/onemkl_dft_mklcpu.hpp"
+#endif
+#ifdef ENABLE_MKLGPU_BACKEND
+#include "oneapi/mkl/dft/detail/mklgpu/onemkl_dft_mklgpu.hpp"
+#endif
+#include "oneapi/mkl/dft/detail/dft_loader.hpp"
+
+namespace oneapi {
+namespace mkl {
+namespace dft {
+
+#ifdef BUILD_RUN
+template <precision prec, domain dom>
+void descriptor<prec, dom>::commit(sycl::queue &queue) {
+    queue_ = queue;
+    pimpl_.reset(detail::create_commit(*this));
+}
+#endif
+
+#ifdef BUILD_COMP
+#ifdef ENABLE_MKLCPU_BACKEND
+template <precision prec, domain dom>
+void descriptor<prec, dom>::commit(backend_selector<backend::mklcpu> selector) {
+    queue_ = selector.get_queue();
+    pimpl_.reset(mklcpu::create_commit(*this));
+}
+#endif
+
+#ifdef ENABLE_MKLGPU_BACKEND
+template <precision prec, domain dom>
+void descriptor<prec, dom>::commit(backend_selector<backend::mklgpu> selector) {
+    queue_ = selector.get_queue();
+    // pimpl_.reset(mklgpu::create_commit(*this));
+}
+#endif
+#endif
+
+// impliment error class
+template <precision prec, domain dom>
+void descriptor<prec, dom>::set_value(config_param param, ...) {
+    int err = 0;
+    va_list vl;
+    va_start(vl, param);
+    switch (param) {
+        case config_param::INPUT_STRIDES: [[fallthrough]];
+        case config_param::OUTPUT_STRIDES: {
+            int64_t *strides = va_arg(vl, int64_t *);
+            if (strides == nullptr)
+                break;
+            if (param == config_param::INPUT_STRIDES)
+                std::copy(strides, strides + rank_ + 1, std::back_inserter(values_.input_strides));
+            if (param == config_param::OUTPUT_STRIDES)
+                std::copy(strides, strides + rank_ + 1, std::back_inserter(values_.output_strides));
+        } break;
```

Comment:
Does this need the `{` and `}` containing the whole case?

Reply:
these make sure that strides is strictly in scope

Reply:
I see your point - I don't think I've seen this before! I think its funny formatting by clang-format too. 

I think leave it as-is then.